res/controllers/Hercules_DJControl_Jogvision-scripts.js

```diff
@@ -0,0 +1,262 @@
+///////////////////////////////////////////////////////////////////////////////
+// JSHint configuration                                                      //
+///////////////////////////////////////////////////////////////////////////////
+/* global engine                                                             */
+/* global print                                                              */
+/* global midi                                                               */
+/////////////////////////////////////////////////////////////////////////////// 
+// ****************************************************************************
+// * Mixxx mapping script file for the Hercules DJControl Jogvision.
+// * Author: DJ Phatso, contributions by Kerrick Staley and David TV
+// * Version 1.3 (November 2019)
+// * Version 1.2 (November 2019)
+// * Version 1.1 (March 2019)
+// * Forum: https://www.mixxx.org/forums/viewtopic.php?f=7&t=12580
+// * Wiki: https://www.mixxx.org/wiki/doku.php/hercules_dj_control_jogvision
+//
+// Changes to v1.3
+// - Enabled the creation of beatloop (beatloop_activate) by using SHIFT+LoopON
+// - Changed "LOOP SIZE" to adjust the beatjump_size (you still can change size
+//   with surrounding buttons)
+// - Added SHIFT+"LOOP SIZE" to move loop left or right by N beats (beatjump)
+//
+// Changes to v1.2
+// - Enabled Jogwheel Outer LED rotation
+// - Enabled Beat LEDs
+// 
+// Changes to v1.1
+// - Controller knob/slider values are queried on startup, so MIXXX is synced.
+// - Fixed vinyl button behavior the first time it's pressed.
+// 
+// v1.0 : Original release
+//
+// ****************************************************************************
+
+var on = 0x7F;
+var off = 0x00;
+var alpha = 1.0 / 8;
+var beta = alpha / 16;
+var ledRotationSpeed = 4;
+var ledRotationTimer;
+
+var DJCJV = {};
+
+DJCJV.scratchButtonState = true;
```

Comment:
This variable name is a bit non-descriptive, what do you think about renaming it to `scratchButtonEnabled`?